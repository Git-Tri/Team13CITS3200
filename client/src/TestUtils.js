const trueErrorLog = console.error;

/**
 * okay this is a weird one to explain
 * basically there an issue with our dev setup (web-pack/react/create react app)
 * where if you test when a component fails it will spam the error log with
 * a message telling you the component failed 
 * since we are testing that, this is completely unneccessary
 * as such i am filtering the console.error to exclude these messages 
 * to give a more readable test output
 * using monkey patching: https://davidwalsh.name/monkey-patching
 */
export function filterConsoleError()
{

    
    console.error = (e) => 
    {

        
        if(typeof(e) === "string" && ! e.includes("React will try to recreate this component tree from scratch using the error boundary you provided"))
        {

            trueErrorLog(e);
        
        }
        else if(typeof(e) !== "string")
        {

            trueErrorLog(e);

        }
        
    }

}

export function unfilterConsoleError()
{

    console.error = trueErrorLog;

}