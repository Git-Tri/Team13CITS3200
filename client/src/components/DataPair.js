
import React from 'react';

/**
 * Creates a simple data pair with a label and a text valued
 * passed into the function
 * @param {*} props the props passed in, must contain a label and text properties 
 */
function DataPair(props)
{

    if(props.text === undefined || props.label === undefined)
    {

        throw new Error("text and label must be defined")

    }

    return(
    <div style={{padding:"0px 15px 0px 15px"}}>
        <b style={{padding:"0px 0px 0px 15px"}}> {props.label}:</b>{props.text}        
    </div>);

}

export default DataPair;