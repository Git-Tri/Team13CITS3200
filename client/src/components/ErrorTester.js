import React from "react";

/**
 * Catches an error and uses the onError callback to allow tests to pass
 * this component is meant for testing purposes only
 * when testing place the error component as a child to this component 
 * then pass in an onError prop or check the state of this component
 * derived from: https://codepen.io/gaearon/pen/wqvxGa?editors=0010
 */
class ErrorTester extends React.Component {
    constructor(props) {
      super(props);
      this.state = { error: null, errorInfo: null };
    }
    
    componentDidCatch(error, errorInfo) {
      this.setState({
        error: error,
        errorInfo: errorInfo
      })

      this.props.onError(error,errorInfo);
    }
    
    render() {
      if (this.state.errorInfo) {
        return (
          <p> Good news an error occurred</p>
          );
      }
      return this.props.children;
    }  
  }

  export default ErrorTester;
