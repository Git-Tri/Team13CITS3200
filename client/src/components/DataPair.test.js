import DataPair from "./DataPair";
import renderer from 'react-test-renderer';
import React from 'react';
import {filterConsoleError,unfilterConsoleError} from "../TestUtils";
import {shallow, mount} from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";
import ErrorTester from "./ErrorTester"

configure({ adapter: new Adapter() });


describe("Data Pair Tests",function()
{


    beforeAll(() =>
    {

        return new Promise(function(resolve,reject)
        {

            filterConsoleError();               

            resolve();

        })

    });

    test("Should throw error when given undefined label",(done) => 
    {

    renderer.create(<ErrorTester              
             onError={(e,em) => {
                    expect(e.message).toEqual("text and label must be defined"); 
                    done();}
                    }>
                 <DataPair text="hello" />
             </ErrorTester>
            );       

    });

    test("Should throw error when given undefined text",(done) => 
    {

    renderer.create(<ErrorTester              
             onError={(e,em) => {
                    expect(e.message).toEqual("text and label must be defined"); 
                    done();}
                    }>
                 <DataPair label="world"/>
             </ErrorTester>
            );       

    });

    test("Should render correctly",() => 
    {

        let wrapper = renderer.create(<DataPair label="hello" text="world"/>)

        expect(wrapper).toMatchSnapshot();

    })



    afterAll(() =>
    {

        return new Promise(function(resolve,reject)
        {

            unfilterConsoleError();

            resolve();

        })

    });
})