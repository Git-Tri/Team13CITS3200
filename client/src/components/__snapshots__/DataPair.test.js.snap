// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Pair Tests Should render correctly 1`] = `
<div
  style={
    Object {
      "padding": "0px 15px 0px 15px",
    }
  }
>
  <b
    style={
      Object {
        "padding": "0px 0px 0px 15px",
      }
    }
  >
     
    hello
    :
  </b>
  world
</div>
`;
